### PR TITLE
Fix a broken link and check internal links

### PR DESCRIPTION
**Pull request by jond01**
In #2153 I aimed to fix a link but broke it.
In this PR, I fix it and add [`mdbook-linkcheck`](https://github.com/Michael-F-Bryan/mdbook-linkcheck) to avoid future cases.

Some past fixes that could have been prevented, in addition to mine in this PR:
* #811
* #2064
* #2146

Note:  
`mdbook-linkcheck` may also check external links with a configuration change.  
It can be beneficial to check also external links from time to time. I ran it here and found 3 broken links.

Maintainers - sorry for the lack of a preceding issue. We can discuss it here.
Some remaining work is to fix the outdated internal links in the translations, not done here.
Let me know what you think about the proposed contribution.

This PR completes #1502.

### CODE REVIEW

**Comment by jond01** [comment]
All the checks pass - see https://github.com/jond01/comprehensive-rust/pull/1.

**Review by djmitche — COMMENTED** [review]
This looks good to me. I think there was some conversation about link-checking before that suggested this might not be a good solution - @mgeisler do you know the details?

**Comment by mgeisler** [comment]
Thanks @jond01, this is an amazing fix! I can't believe we had so many broken links :grimacing: 

> This looks good to me. I think there was some conversation about link-checking before that suggested this might not be a good solution - @mgeisler do you know the details?

Yeah, this was #1502, which @jond01 also found — I think the only problem back then was that this was also executed for the translations and that they had a bunch of errors.

So I think fixing this for the English version first is a great idea. We can let the translators know and let them fix the translations one by one. We could even let them opt into strict link checking once their language has no errors.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -147,6 +147,10 @@ jobs:
           MDBOOK_OUTPUT='{"xgettext": {"pot-file": "messages.pot"}}' mdbook build -d po
           msgfmt -o /dev/null --statistics po/messages.pot
 
+      - name: Install mdbook-linkcheck
+        if: matrix.language == 'en'
```
[the remark this answers is not in the document]
Comment (mgeisler) [reply .github/workflows/build.yml:151]: I would suggest adding a comment here explaining why the tool is only installed for English right now.

[the remark this answers is not in the document]
Comment (jond01) [reply .github/workflows/build.yml]: Good idea, done in 22a274b.

**Review by mgeisler — APPROVED** [review]
Thanks a lot!

**Comment by mgeisler** [comment]
I opened #2247 to track a small followup problem: the build is broken for all languages except English :smile: The problem is very likely related to caching and non-hermetic builds.

**Comment by mgeisler** [comment]
This also fixed #957.

**Pull request merged**